cpp/include/cudf/detail/reduction_operators.cuh

```diff
@@ -88,13 +88,23 @@ struct simple_op {
    *
    * @return element transformer functor object
    */
-  template <typename ResultType>
+  template <typename ResultType,
+            typename std::enable_if_t<!is_template_of<thrust::pair, ResultType>::value>* = nullptr>
   auto get_element_transformer()
   {
     using element_transformer = typename Derived::transformer<ResultType>;
     return element_transformer{};
   }
 
+  template <typename ResultType,
+            typename std::enable_if_t<is_template_of<thrust::pair, ResultType>::value>* = nullptr>
+  auto get_element_transformer()
```

Comment:
Is there a reason why this is a template specialization instead of a separate method like `get_null_replacing_element_transformer()`?